src/text.tsx

```diff
@@ -6,23 +6,90 @@ export interface Tag {
   end: string;
 }
 
-export interface Interpolations {
+export interface Suffix {
+  begin: string;
+  end: string;
+}
+
+interface SingleInterpolations {
   [keyof: string]: React.ReactNode;
 }
 
+interface PairInterpolations {
+  [keyof: string]: (match: React.ReactNode) => React.ReactNode;
+}
+export interface Interpolations {
+  [keyof: string]: SingleInterpolations[string] | PairInterpolations[string];
+}
+
 export interface TextProps {
   id: string;
   interpolations?: Interpolations;
   tag?: Tag;
+  suffix?: Suffix;
+}
+
+function isFunction(val: Interpolations[string]): val is PairInterpolations[string] {
+  return typeof val === "function";
 }
 
-export const Text = ({ id, interpolations, tag }: TextProps) => {
+const toTag = (key: string, tag: Tag, suffix: string = '') =>
+  `${tag.start}${key}${suffix}${tag.end}`;
+
+const escapeRegex = (str: string) => str.replace(/([.?*+^$[\]\\(){}|-])/g, '\\$1');
+
+const nodeRegex = (begin: string, end: string) =>
+  new RegExp(`${escapeRegex(begin)}(.*)${escapeRegex(end)}`, 'g');
```

Comment:
```suggestion
  new RegExp(`${escapeRegex(begin)}(.*)${escapeRegex(end)}`, 'gi');
```

Would this make it case-insensitive?

Reply:
Needed an additional change 😄